ext/dapr-ext-workflow/dapr/ext/workflow/aio/workflow_runtime.py

```diff
@@ -0,0 +1,149 @@
+from dapr.ext.workflow.workflow_activity_context import Activity
+from dapr.ext.workflow.workflow_runtime import WorkflowRuntime as WorkflowRuntimeSync
+
+T = TypeVar('T')
+TInput = TypeVar('TInput')
+TOutput = TypeVar('TOutput')
+
+ClientInterceptor = Union[
+    grpc.aio.UnaryUnaryClientInterceptor,
+    grpc.aio.UnaryStreamClientInterceptor,
+    grpc.aio.StreamUnaryClientInterceptor,
+    grpc.aio.StreamStreamClientInterceptor,
+]
+
+
+class WorkflowRuntime(WorkflowRuntimeSync):
+    """WorkflowRuntime is the entry point for registering workflows and async activities."""
+
+    def __init__(
+        self,
+        host: Optional[str] = None,
+        port: Optional[str] = None,
+        logger_options: Optional[LoggerOptions] = None,
+        interceptors: Optional[Sequence[ClientInterceptor]] = None,
+        maximum_concurrent_activity_work_items: Optional[int] = None,
+        maximum_concurrent_orchestration_work_items: Optional[int] = None,
+        maximum_thread_pool_workers: Optional[int] = None,
+        main_event_loop: Optional[asyncio.AbstractEventLoop] = None,
+    ):
+        super().__init__(
+            host,
+            port,
+            logger_options,
+            interceptors,
+            maximum_concurrent_activity_work_items,
+            maximum_concurrent_orchestration_work_items,
+            maximum_thread_pool_workers,
+        )
+        self._main_event_loop = main_event_loop
+        self._logger = Logger('WorkflowRuntime', logger_options)
+
+    def register_activity(self, fn: Activity, *, name: Optional[str] = None):
+        """Registers an async workflow activity and ensures proper execution and metadata.
+        This mirrors the decorator behavior so direct registration behaves identically.
+        """
+        # Validate/prepare alternate name on the original function first (not the wrapper)
+        if hasattr(fn, '_activity_registered'):
+            alt_name = fn.__dict__['_dapr_alternate_name']
+            raise ValueError(f'Activity {fn.__name__} already registered as {alt_name}')
+        if hasattr(fn, '_dapr_alternate_name'):
+            alt_name = fn._dapr_alternate_name
+            if name is not None:
+                raise ValueError(f'Activity {fn.__name__} already has an alternate name {alt_name}')
+        else:
+            alt_name = name if name else fn.__name__
+            fn.__dict__['_dapr_alternate_name'] = alt_name
+
+        # Build the target function that super().register_activity will wrap.
+        if self._main_event_loop:
+
+            @wraps(fn)
+            def target_fn(*args, **kwargs):
+                result = fn(*args, **kwargs)
+                if inspect.isawaitable(result):
+                    future = asyncio.run_coroutine_threadsafe(result, self._main_event_loop)
+                    return future.result()
+                return result
+        else:
+
+            @wraps(fn)
+            def target_fn():
+                return fn
```

Comment:
1. Does this mean that if the `main_event_loop` was not provided in the constructor, we'll silently treat all activities as synchronous? If this is the case, should we make `main_event_loop` required? Otherwise, it seems to be defeating the purpose. Or am I missing something?
2. Isn't it supposed to call `fn()` instead of returning the function itself?